security-and-privacy-questionnaire-unrestricted-webusb.md

```diff
@@ -0,0 +1,125 @@
+# [Self-Review Questionnaire: Security and Privacy](https://w3ctag.github.io/security-questionnaire/)
+
+[Markdown template](https://raw.githubusercontent.com/w3ctag/security-questionnaire/main/questionnaire.markdown)
+
+> 01.  What information does this feature expose,
+>      and for what purposes?
+
+(For WebUSB in general:) By default, the API exposes no information about a user's connected devices. Once the user grants permission for a site to access a device, the API exposes details about the device including device identifiers like the serial number string, vendor/product IDs, product name string, device firmware version, USB device class, supported configurations and interfaces.
+
+Applications use the vendor ID, product ID, and firmware version to identify the connected device model and determine compatibility.
+
+The serial number string is used to identify a specific device and distinguish the device from other devices with the same vendor and product IDs.
+
+The product name string is used as a human-facing identifier so that the application can present the device to the user with the same name that is used in permissions surfaces.
+
+The device's supported configurations and interfaces describe how to communicate with the device. Applications use this information to deduce the device's capabilities, compose transfers to be sent to the device, and interpret transfers received from the device.
+
+By allowing a site to communicate with a peripheral device it is exposed to any information that the device makes available through that connection. Information communicated with the device may include device identifiers that can be used for fingerprinting, private user data, security credentials, and other sensitive data. WebUSB [restricts the classes of devices](https://wicg.github.io/webusb/#has-a-protected-interface-class) that can be accessed through the API to limit the potential for abuse.
+
+(For Unrestricted WebUSB:) The Unrestricted WebUSB feature removes the restrictions on protected device classes so that trusted applications can access these devices.
+
+> 02.  Do features in your specification expose the minimum amount of information
+>      necessary to implement the intended functionality?
+
+Yes, an application with the `"usb-unrestricted"` permission receives the same information that is normally available through WebUSB API, except that the application additionally receives information about protected devices.
```

Comment:
```suggestion
(For WebUSB in general:) Yes, a site only receives information about the device the user has chosen to share with it.

(For Unrestricted WebUSB:) Applications must opt-in to the ability to access otherwise protected device by requesting the `"usb-unrestricted"` permission which may subject them to additional scrutiny through a process external to this API.
```

Reply:
Done